mod-source-record-manager-server/src/main/java/org/folio/services/flowcontrol/RawRecordsFlowControlService.java

```diff
@@ -0,0 +1,33 @@
+package org.folio.services.flowcontrol;
+
+/**
+ * Service to implement flow control logic to be able to import OCLC files in between imports of huge files.
+ * Its necessary to not wait for importing 1 record file until other big files in progress.
+ * Its can be suitable not only for OCLC, but also for other files that skipping DI_RAW_RECORDS_CHUNK_READ stage
+ * and directly pushing messages to DI_RAW_RECORDS_CHUNK_PARSED topic.
+ */
+public interface RawRecordsFlowControlService {
+
+  /**
+   * Tracks each DI_RAW_RECORDS_CHUNK_READ event, this method can also pause processing
+   * of DI_RAW_RECORDS_CHUNK_READ topic when flow control conditions met.
+   *
+   * @param initialRecordsCount records count in the batch
```

Comment:
I am not able to discern what `initialRecordsCount` is pointing to OR how it will play with `trackchinkReceivedEvent`. The method doesnt accept a batch. Where is the batch coming from?

When the questions above are answered, it could help me to know what additional comments could be added here to make this interface self-documenting.

Reply:
The latest refactor of this interface has cleared things for me.